cmd/zed/manage/lakemanager/scan.go

```diff
@@ -0,0 +1,64 @@
+package lakemanager
+
+import (
+	"context"
+	"time"
+
+	"github.com/brimdata/zed/lake/data"
+	"github.com/brimdata/zed/lake/pools"
+	"github.com/brimdata/zed/order"
+	"github.com/brimdata/zed/runtime/expr/extent"
+)
+
+type ObjectReader interface {
+	Next() (*data.Object, error)
```

Comment:
Nit: Since this method is called Next rather than Read, maybe call the interface ObjectIterator?